app/org/sagebionetworks/bridge/dynamodb/DynamoUserConsentDao.java

```diff
@@ -93,19 +119,63 @@ public long getNumberOfParticipants(StudyIdentifier studyIdentifier) {
         condition.withAttributeValueList(new AttributeValue().withS(studyIdentifier.getIdentifier()));
         scan.addFilterCondition("studyKey", condition);
 
-        // We need the count of unique study participants, users may end up signing 
+        // We need the count of unique study participants, users may end up signing
         // more than one version of the consent.
         Set<String> healthCodes = Sets.newHashSet();
-        List<DynamoUserConsent2> mappings = mapper.scan(DynamoUserConsent2.class, scan);
-        for (DynamoUserConsent2 consent : mappings) {
-            healthCodes.add(consent.getHealthCode());
-        }
+        List<DynamoUserConsent2> mappings = mapperV2.scan(DynamoUserConsent2.class, scan);
+        mappings.stream().forEach(consent -> healthCodes.add(consent.getHealthCode()));
         return healthCodes.size();
     }
 
-    private DynamoUserConsent2 getUserConsent(String healthCode, StudyConsent studyConsent) {
-        DynamoUserConsent2 consent = new DynamoUserConsent2(healthCode, studyConsent);
-        consent = mapper.load(consent);
-        return consent;
+    @Override
+    public void deleteConsentRecords(String healthCode, StudyIdentifier studyIdentifier) {
+        List<DynamoUserConsent3> consentsV3 = getAllUserConsentRecords3(healthCode, studyIdentifier.getIdentifier());
+        if (!consentsV3.isEmpty()) {
+            List<FailedBatch> failures = mapperV3.batchDelete(consentsV3);
+            BridgeUtils.ifFailuresThrowException(failures);
+        }
+        DynamoUserConsent2 consentV2 = getUserConsent2(healthCode, studyIdentifier.getIdentifier());
+        if (consentV2 != null) {
+            mapperV2.delete(consentV2);    
+        }
```

Comment:
As written, if this code fails to delete v3 consents, it will skip deleting v2 consents. I know this is only used for testing at the moment, but is this okay?